internal/scan-manager/scanners/veracode/src/main/java/org/wso2/security/tools/scanmanager/scanners/veracode/service/VeracodeScanner.java

```diff
@@ -107,41 +105,51 @@ private static void loadConfiguration() throws IOException {
      * Run the scan using product zip file.
      *
      * @param scanRequest Object that represent the required information for the scanner operation
-     * @return ResponseEntity with status of the request
      */
     @Override
-    public ResponseEntity startScan(ScannerScanRequest scanRequest) {
+    public void startScan(ScannerScanRequest scanRequest) {
         scanContext.setJobId(scanRequest.getJobId());
         scanContext.setAppId(scanRequest.getAppId());
         scanContext.setArtifactLocation(scanRequest.getFileMap().get(VeracodeScannerConstants.SCAN_ARTIFACT).get(0));
-        ResponseEntity responseEntity;
 
         if (scanRequest.getFileMap().get(VeracodeScannerConstants.SCAN_ARTIFACT) != null) {
             if (!StringUtils.isEmpty(scanContext.getAppId())) {
-                responseEntity = new ResponseEntity<>(HttpStatus.ACCEPTED);
-                startVeracodeScan();
+                if (!Thread.currentThread().isInterrupted()) {
```

Comment:
Doing a not operation at the if condition is little more costlier. We can write `if(interrupted)` condition first and then do the actual work in the `else` for better performance. This performance gain is not significant for our case, but it's better to follow that as a practice whenever that makes sense.